bee-protocol/src/worker/broadcaster.rs

```diff
@@ -32,12 +34,31 @@ pub(crate) struct BroadcasterWorkerEvent {
 
 pub(crate) struct BroadcasterWorker {
     network: Network,
-    receiver: Receiver,
+}
+
+#[async_trait]
+impl Worker for BroadcasterWorker {
+    const DEPS: &'static [TypeId] = &[];
+
+    type Event = BroadcasterWorkerEvent;
+    type Receiver = ShutdownStream<Fuse<mpsc::UnboundedReceiver<BroadcasterWorkerEvent>>>;
```

Comment:
I think I mentioned this before, but I'd really like to replace all of this sort of thing with a central event bus + a way to get references to other workers via the `N: Node`. Probably for a follow-up PR though.